xclim/indices/_threshold.py

```diff
@@ -2107,6 +2108,56 @@ def tropical_nights(
     return tn_days_above(tasmin, thresh=thresh, freq=freq)
 
 
+@declare_units(pr="[precipitation]", prc="[precipitation]", thresh="[precipitation]")
+def rprctot(
+    pr: xarray.DataArray,
+    prc: xarray.DataArray,
+    thresh: str = "1.0 mm/day",
+    freq: str = "YS",
+) -> xarray.DataArray:
+    """Proportion of accumulated precipitation arising from convective processes.
+
+    Return the proportion of total accumulated precipitation due to convection on days with total precipitation exceeding a specified threshold during the given period.
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+      Daily precipitation.
+    prc : xarray.DataArray
+      Daily convective precipitation.
+    thresh : str
+      Precipitation value over which a day is considered wet.
+    freq : str
+      Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray, [time]
+      The proportion of the total precipition accounted for by convective precipitation for each period [1].
+
+    Examples
+    --------
+    The following would compute for each grid cell of file `pr.day.nc` the proportion of total rainfall
+    arising from convective precipitation on days with more than 5mm rainfall at the seasonal frequency, ie DJF, MAM, JJA, SON, DJF, etc.:
+
+    >>> from xclim.indices import rprctot
+    >>> pr = xr.open_dataset(path_to_pr_file).pr
+    >>> prc = xr.open_dataset(path_to_prc_file).prc
+    >>> wd = rprctot(pr, prc, thresh="5 mm/day", freq="QS-DEC")
+    """
+
+    thresh = convert_units_to(thresh, pr, "hydro")
```

Comment:
```suggestion
    thresh = convert_units_to(thresh, pr, "hydro")
    prc = convert_units_to(prc, pr)
```
To ensure the ratio is taken with the same units!